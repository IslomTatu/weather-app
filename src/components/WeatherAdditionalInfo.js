import dayjs from 'dayjs'
import React from 'react'

import styles from './styles.module.scss'

export const WeatherAdditionalInfo = ({ hourly = [] }) => {
  const dt = hourly ? hourly[0]?.dt : new Date().getTime()

  return (
    <div className={styles.table}>
      <div className={styles['additional-info-header']}>
        <div className={styles.title}>
          <h1>
            {dayjs.unix(dt).format('DD')}{' '}
            <span>{dayjs.unix(dt).format('MMMM')}</span>
          </h1>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Min &#8451;</span>
          </div>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Max &#8451;</span>
          </div>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Pressure, mmHg</span>
          </div>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Humidity (%)</span>
          </div>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Wind gust</span>
          </div>
        </div>
        <div className={styles['additional-info-item']}>
          <div className={styles['additional-info-item-text']}>
            <span>Wind speed (m/s)</span>
          </div>
        </div>
      </div>
      {hourly && hourly.map(item => {
        const { main, wind } = item
        return (
          <div key={item.dt} className={styles['additional-info-body']}>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{dayjs.unix(item?.dt).format('HH:MM')}</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{main.temp_min}&#176;</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{main.temp_max}&#176;</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{main?.pressure}</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{main?.humidity}</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{wind?.gust}</span>
              </div>
            </div>
            <div className={styles['additional-info-item']}>
              <div className={styles['additional-info-item-value']}>
                <span>{wind?.speed}</span>
              </div>
            </div>
          </div>
        )
      })}
    </div>
  )
}
