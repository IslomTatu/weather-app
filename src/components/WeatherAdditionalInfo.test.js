/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'

import { WeatherAdditionalInfo } from './WeatherAdditionalInfo'

describe('WeatherAdditionalInfo component renders testing', () => {
  it('WeatherAdditionalInfo renders without crashing', () => {
    expect(shallow(<WeatherAdditionalInfo />)).toMatchSnapshot()
  })

  it('WeatherAdditionalInfo renders hourly data empty array', () => {
    const hourly = []
    expect(shallow(<WeatherAdditionalInfo hourly={hourly} />))
  })

  it('Check component if hourly props null or undefined', () => {
    expect(shallow(<WeatherAdditionalInfo hourly={undefined} />))
    expect(shallow(<WeatherAdditionalInfo hourly={null} />))
  })
})
