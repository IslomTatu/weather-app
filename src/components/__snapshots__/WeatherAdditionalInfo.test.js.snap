// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherAdditionalInfo component renders testing WeatherAdditionalInfo renders without crashing 1`] = `ShallowWrapper {}`;
